### PR TITLE
add Status check assertions for repair_test

### PR DESCRIPTION
**Pull request by ajkr**

### CODE REVIEW

File: db/db_test_util.cc [hunk 1]
```diff
@@ -1407,29 +1407,33 @@ void DBTestBase::CopyFile(const std::string& source,
...
-      env_->GetFileSize(file_path, &file_size);
-      res[file_path] = file_size;
-      if (total_size) {
-        *total_size += file_size;
+  Status s = env_->GetChildren(dbname_, &files);
+  if (s.ok()) {
+    for (auto& file_name : files) {
+      uint64_t number;
+      FileType type;
+      std::string file_path = dbname_ + "/" + file_name;
```
Comment (riversand963) [inline db/db_test_util.cc:1422]: We need to construct the `file_path` only if the check on line 1423 is true.

**Review by riversand963 — APPROVED** [review]
LGTM.

File: db/db_test_util.cc (hunk 1 above)
Comment (ajkr) [reply db/db_test_util.cc]: Fixed.

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

File: db/table_cache.cc [hunk 2]
```diff
@@ -171,7 +171,8 @@ Status TableCache::FindTable(const ReadOptions& ro,
 
   if (*handle == nullptr) {
     if (no_io) {  // Don't do IO and return a not-found status
-      return Status::Incomplete("Table not found in table_cache, no_io is set");
+      s = Status::Incomplete("Table not found in table_cache, no_io is set");
```
Comment (mrambacher) [inline db/table_cache.cc:174]: Please fix the comment on the earlier line

File: db/repair.cc [hunk 3]
```diff
@@ -163,31 +164,40 @@ class Repairer {
...
   }
 
-  ~Repairer() {
-    if (db_lock_ != nullptr) {
-      env_->UnlockFile(db_lock_);
+  Status Close() {
+    Status s = Status::OK();
+    if (!closed_) {
+      if (db_lock_ != nullptr) {
+        s = env_->UnlockFile(db_lock_);
```
Comment (mrambacher) [inline db/repair.cc:171]: What happens if the Unlock fails?  The file will be marked as Closed but the db_lock_ is still hanging out?

File: db/table_cache.cc (hunk 2 above)
Comment (ajkr) [reply db/table_cache.cc:174]: Deleted the comment since it was just inaccurately repeating the code.

File: db/repair.cc (hunk 3 above)
Comment (ajkr) [reply db/repair.cc:171]: Fixed to set to nullptr since `UnlockFile()` is not retryable and deletes the pointer passed to it even in the failure case.

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

File: db/table_cache.cc [hunk 4]
```diff
@@ -170,8 +170,9 @@ Status TableCache::FindTable(const ReadOptions& ro,
                            const_cast<bool*>(&no_io));
 
   if (*handle == nullptr) {
-    if (no_io) {  // Don't do IO and return a not-found status
-      return Status::Incomplete("Table not found in table_cache, no_io is set");
+    if (no_io) {
+      s = Status::Incomplete("Table not found in table_cache, no_io is set");
```
Comment (ajkr) [inline db/table_cache.cc:174]: Appveyor says:

```
    35>c:\projects\rocksdb\db\table_cache.cc(173): error C2065: 's': undeclared identifier [C:\projects\rocksdb\build\rocksdb.vcxproj]
    35>c:\projects\rocksdb\db\table_cache.cc(174): error C2065: 's': undeclared identifier [C:\projects\rocksdb\build\rocksdb.vcxproj]
```

I wonder why it thinks `s` is undeclared...

Comment (ajkr) [reply db/table_cache.cc:174]: It must have automatically rebased without any conflict and lost the declaration. I've never seen a clean rebase cause a problem before.

**Pull request closed without merging**